src/bootstrapper.cc

```diff
@@ -61,33 +62,41 @@ void PromiseRejectCallback(PromiseRejectMessage message) {
   PromiseRejectEvent event = message.GetEvent();
 
   Environment* env = Environment::GetCurrent(isolate);
-  Local<Function> callback;
+  Local<Function> callback = env->promise_handler_function();
   Local<Value> value;
+  Local<Value> type;
 
   if (event == v8::kPromiseRejectWithNoHandler) {
-    callback = env->promise_reject_unhandled_function();
     value = message.GetValue();
-
-    if (value.IsEmpty())
-      value = Undefined(isolate);
-
+    type = Number::New(env->isolate(), 0);
     unhandledRejections++;
+    TRACE_COUNTER2(TRACING_CATEGORY_NODE2(promises, rejections),
+                  "rejections",
+                  "unhandled", unhandledRejections,
+                  "handledAfter", rejectionsHandledAfter);
   } else if (event == v8::kPromiseHandlerAddedAfterReject) {
-    callback = env->promise_reject_handled_function();
     value = Undefined(isolate);
-
+    type = Number::New(env->isolate(), 1);
     rejectionsHandledAfter++;
+    TRACE_COUNTER2(TRACING_CATEGORY_NODE2(promises, rejections),
+                  "rejections",
+                  "unhandled", unhandledRejections,
+                  "handledAfter", rejectionsHandledAfter);
+  } else if (event == v8::kPromiseResolveAfterResolved) {
+    value = message.GetValue();
+    type = Number::New(env->isolate(), 2);
+  } else if (event == v8::kPromiseRejectAfterResolved) {
+    value = message.GetValue();
+    type = Number::New(env->isolate(), 3);
   } else {
-    return;
+    UNREACHABLE();
```

Comment:
Keeping `return;` allows V8 to add new events without breaking Node

Reply:
The reason I changed it was to actually detect the new events up front. I'll change it back to return tough.